auth/router.js

```diff
@@ -0,0 +1,48 @@
+const { Router } = require("express");
+const { toJWT } = require("./jwt");
+const { User } = require("../models");
+const bcrypt = require("bcrypt");
+
+async function login(res, next, name = null, password = null) {
+  if (!name || !password) {
+    res.status(400).send({
+      message: "Please supply a valid name and password"
+    });
+  } else {
+    try {
+      const user = await User.findOne({ where: { name: name } });
+      if (!user) {
+        res.status(400).send({
+          message: "User with that name does not exist"
+        });
+      }
+      // 2. use bcrypt.compareSync to check the password against the stored hash
+      else if (bcrypt.compareSync(password, user.password)) {
```

Comment:
by @akellbl4:

I don't remember when I used bcrypt last time but I think it possible to use async compare in this place.
I can suggest using only unblocking operations in runtime.
You can learn why it is bad to use a blocking operation by articles and videos about the event loop.
I like this one https://www.youtube.com/watch?v=cCOL7MC4Pl0 it more about browser but v8 works in the same way in node just without rendering.